osu.Game.Rulesets.Osu/OsuInputManager.cs

```diff
@@ -43,17 +53,52 @@ protected override bool Handle(UIEvent e)
             return base.Handle(e);
         }
 
+        private OsuAction getActionForTouchSource(TouchSource source)
+        {
+            int sourceNumber = numberedTouchSources[source];
+            return sourceNumber % 2 == 0 ? OsuAction.RightButton : OsuAction.LeftButton;
+        }
+
         protected override bool HandleMouseTouchStateChange(TouchStateChangeEvent e)
         {
-            if (!AllowUserCursorMovement)
+            var source = e.Touch.Source;
+
+            int touchNumber = numberedTouchSources[source];
+
+            if (touchNumber > simultaneous_touches_limit)
+                return false;
```

Comment:
`TouchSource` is made in a way that allows mapping sources to integers without the need of any extra object or explicit dictionaries.
```suggestion
            var source = e.Touch.Source;
            if (source - TouchSource.Touch1 >= simultaneous_touches_limit)
                return false;
```